extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,1094 @@
+		array( 'table' => 'ipblocks', 'userid_column' => 'ipb_by', 'username_column' => 'ipb_by_text' ),
+		array( 'table' => 'ipblocks', 'userid_column' => 'ipb_user', 'username_column' => 'ipb_address' ),
+		array( 'table' => 'logging', 'userid_column' => null, 'username_column' => 'log_title',
+			'conds' => array(
+				'log_namespace' => NS_USER,
+			) ),
+		array( 'table' => 'oldimage', 'userid_column' => 'oi_user', 'username_column' => 'oi_user_text' ),
+		array( 'table' => 'recentchanges', 'userid_column' => 'rc_user', 'username_column' => 'rc_user_text' ),
+		array( 'table' => 'revision', 'userid_column' => 'rev_user', 'username_column' => 'rev_user_text' ),
+		# disable in 1.19 array( 'table' => 'text', 'userid_column' => 'old_user', 'username_column' => 'old_user_text' ),
+		array( 'table' => 'user_newtalk', 'userid_column' => null, 'username_column' => 'user_ip' ),
+		# Core 1.16 tables
+		array( 'table' => 'logging', 'userid_column' => 'log_user', 'username_column' => 'log_user_text' ),
+
+		# Template entry
+//		array( 'table' => '...', 'userid_column' => '...', 'username_column' => '...' ),
+	);
+
+	/**
+	 * Stores the predefined tasks to do for every local wiki database for IP addresses.
+	 * Here should be mentioned all core tables not connected to any extension.
+	 *
+	 * @var $mLocalIpDefaults array
+	 */
+	static private $mLocalIpDefaults = [
+		[ 'table' => 'archive', 'userid_column' => 'ar_user', 'username_column' => 'ar_user_text' ],
+		[ 'table' => 'filearchive', 'userid_column' => 'fa_user', 'username_column' => 'fa_user_text' ],
+		[ 'table' => 'ipblocks', 'userid_column' => 'ipb_user', 'username_column' => 'ipb_address' ],
+		[ 'table' => 'recentchanges', 'userid_column' => 'rc_user', 'username_column' => 'rc_user_text' ],
+		[ 'table' => 'revision', 'userid_column' => 'rev_user', 'username_column' => 'rev_user_text' ],
+	];
+
+	private $mRequestData = null;
+	private $mActionConfirmed = false;
+
+	private $mOldUsername = '';
+	private $mNewUsername = '';
+	private $mUserId = 0;
+	private $mFakeUserId = 0;
+	private $mPhalanxBlockId = 0;
+	private $mRequestorId = 0;
+	private $mRequestorName = '';
+	private $mReason = null;
+	private $mNotifyUser;
+
+	private $mErrors = array();
+	private $mWarnings = array();
+
+	private $mInternalLog = '';
+
+	private $mLogDestinations = array(
+		array( self::LOG_STANDARD, null ),
+	);
+	/**
+	 *
+	 * @var BatchTask
+	 */
+	private $mLogTask = null;
+
+	/**
+	 * @var string
+	 */
+	private $mUserRenameTaskId = null;
+
+	/**
+	 * Creates new rename user process
+	 *
+	 * @param string $oldUsername Old username
+	 * @param string $newUsername New username
+	 * @param bool $confirmed Has the user confirmed all the warnings he got?
+	 * @param string $reason
+	 * @param bool $notifyUser Whether to notify the renamed user
+	 */
+	public function __construct( $oldUsername, $newUsername, $confirmed = false, $reason = null, $notifyUser = true ) {
+		global $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)

[project-id]: wikia:app